include/openthread-types.h

```diff
@@ -336,6 +341,30 @@ typedef enum
 } otDeviceRole;
 
 /**
+ * This structure represents the MAC layer counter.
+ */
+typedef struct otMacCounter
+{
+    uint32_t mTx;                  ///< The total number of transmissions.
+    uint32_t mTxData;              ///< The number of transmitted data.
+    uint32_t mTxDataPoll;          ///< The number of transmitted data poll.
+    uint32_t mTxBeacon;            ///< The number of transmitted beacon.
+    uint32_t mTxBeaconRequest;     ///< The number of transmitted beacon request.
+    uint32_t mTxRetry;             ///< The number of retransmission times.
+    uint32_t mTxErrAck;            ///< The number of unacked transmissions.
```

Comment:
How does someone determine the number of acked transmissions?  How does someone determine the number of unacked transmissions?  Would it be easier to simply keep a count for acked transmissions and unacked transmissions?  Right now, it seems that `mTxRetry` is the number of retransmissions and `mTxErrAck` is the number of times the MAC has given up.

If these counters are following a some standard convention or model, please let me know.